### PR TITLE
Increase migration performance

### PR DESCRIPTION
**Pull request by Jotschi**

### CODE REVIEW

File: core/src/main/java/com/gentics/mesh/core/data/search/impl/SearchQueueBatchImpl.java [hunk 1]
```diff
@@ -299,15 +305,24 @@ public Completable processAsync() {
...
+			}
+		} else {
+			clear();
 		}
 	}
 
 	@Override
 	public void processSync() {
-		processSync(120, TimeUnit.SECONDS);
+		if (!(searchProvider instanceof DevNullSearchProvider)) {
```
Comment (npomaroli) [inline core/src/main/java/com/gentics/mesh/core/data/search/impl/SearchQueueBatchImpl.java]: Not necessary (already implemented in processSync(long, TimeUnit)

File: core/src/main/java/com/gentics/mesh/core/endpoint/migration/micronode/MicronodeMigrationHandler.java [hunk 2]
```diff
@@ -106,8 +111,19 @@ public Completable migrateMicronodes(Release release, MicroschemaContainerVersio
 					status.commit();
 				}
 			}
+			if (count % 500 == 0) {
```
Comment (npomaroli) [inline core/src/main/java/com/gentics/mesh/core/endpoint/migration/micronode/MicronodeMigrationHandler.java]: Move after count++

File: core/src/main/java/com/gentics/mesh/core/endpoint/migration/release/ReleaseMigrationHandler.java [hunk 3]
```diff
@@ -81,8 +85,19 @@ public void migrateRelease(Release newRelease, MigrationStatusHandler status) {
 					status.commit();
 				}
 			}
+			if (count % 500 == 0) {
```
Comment (npomaroli) [inline core/src/main/java/com/gentics/mesh/core/endpoint/migration/release/ReleaseMigrationHandler.java]: Move after count++

File: core/src/main/java/com/gentics/mesh/core/endpoint/migration/release/ReleaseMigrationHandler.java [hunk 4]
```diff
@@ -65,10 +65,14 @@ public void migrateRelease(Release newRelease, MigrationStatusHandler status) {
...
-			SearchQueueBatch sqb = db.tx(() -> {
-				return migrateNode(node, oldRelease, newRelease);
-			});
+			// Create a new SQB to handle the ES update
+			if (sqb == null) {
+				sqb = searchQueue.create();
+			}
+			migrateNode(node, sqb, oldRelease, newRelease);
+
 			if (sqb != null) {
```
Comment (npomaroli) [inline core/src/main/java/com/gentics/mesh/core/endpoint/migration/release/ReleaseMigrationHandler.java]: Remove this if-block

File: core/src/main/java/com/gentics/mesh/core/endpoint/migration/release/ReleaseMigrationHandler.java [hunk 5]
```diff
@@ -98,46 +113,46 @@ public void migrateRelease(Release newRelease, MigrationStatusHandler status) {
 	 * the tags will be update to correspond with the new release structure.
 	 * 
 	 * @param node
+	 * @param batch
 	 * @param oldRelease
 	 * @param newRelease
 	 * @return
```
Comment (npomaroli) [inline core/src/main/java/com/gentics/mesh/core/endpoint/migration/release/ReleaseMigrationHandler.java]: Remove @return

File: core/src/main/java/com/gentics/mesh/core/data/search/impl/SearchQueueBatchImpl.java (hunk 1 above)
Comment (Jotschi) [reply core/src/main/java/com/gentics/mesh/core/data/search/impl/SearchQueueBatchImpl.java]: Removed

**Pull request merged**